java/ql/src/Security/CWE/CWE-338/JHipsterGeneratedPRNG.qhelp

```diff
@@ -0,0 +1,55 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+<p>This query detects instances of <code>RandomUtil.java</code> generated by a <a href="https://www.jhipster.tech/">JHipster</a> version vulnerable to <a href="https://github.com/jhipster/jhipster-kotlin/security/advisories/GHSA-j3rh-8vwq-wh84">CVE-2019-16303</a>.</p>
+
+<p>Using one password reset token from your app combined with the proof of concept (POC) linked below, an attacker can determine all future password reset tokens to be generated by this server.
+This allows an attacker to pick and choose what account they would like to takeover by sending account password reset requests for targeted accounts.</p>
+
+<p>This vulnerability has a 
+<a href="https://nvd.nist.gov/vuln-metrics/cvss/v3-calculator?name=CVE-2019-16303&amp;vector=AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H&amp;version=3.1&amp;source=NIST">
+  CVSS v3.0 Base Score of 9.8/10
+</a>.</p>
+</overview>
+
+<example>
+
+<p>The example below shows the vulnerable <code>RandomUtil</code> class generated by JHipster.</p>
+<sample src="JHipsterGeneratedPRNGVulnerble.java" />
+
+<p>Below is a fixed version of the <code>RandomUtil</code> class.</p>
+<sample src="JHipsterGeneratedPRNGFixed.java" />
+
+</example>
+
+<recommendation>
+
+<p>An automated refactoring <a href="https://github.com/openrewrite/rewrite">rewrite</a> module <a href="https://github.com/moderneinc/jhipster-cwe-338"> can be found here</a>.</p>
+</recommendation>
+
+<references>
+
+<li>
+  <a href="https://blog.cloudflare.com/why-randomness-matters/">
+    Cloudflare Blog: Why secure systems require random numbers
+  </a>
+</li>
+<li>
+  <a href="https://news.ycombinator.com/item?id=639976">
+    How I Hacked Hacker News (with arc security advisory)
+  </a>
+<li>
```

Comment:
This was hidden in the previous Qhelp preview test run by the other validation errors:

```suggestion
</li>
```